sky/authentication.py

```diff
@@ -195,31 +196,55 @@ def setup_gcp_authentication(config: Dict[str, Any]) -> Dict[str, Any]:
          if item['key'] == 'enable-oslogin'), {}).get('value', 'False')
 
     if project_oslogin.lower() == 'true':
+
         # project.
         logger.info(
             f'OS Login is enabled for GCP project {project_id}. Running '
             'additional authentication steps.')
-        # Read the account information from the credential file, since the user
-        # should be set according the account, when the oslogin is enabled.
-        config_path = os.path.expanduser(clouds.gcp.GCP_CONFIG_PATH)
-        sky_backup_config_path = os.path.expanduser(
-            clouds.gcp.GCP_CONFIG_SKY_BACKUP_PATH)
-        assert os.path.exists(sky_backup_config_path), (
-            'GCP credential backup file '
-            f'{sky_backup_config_path!r} does not exist.')
-
-        with open(sky_backup_config_path, 'r') as infile:
-            for line in infile:
-                if line.startswith('account'):
-                    account = line.split('=')[1].strip()
-                    break
-            else:
-                with ux_utils.print_exception_no_traceback():
-                    raise RuntimeError(
-                        'GCP authentication failed, as the oslogin is enabled '
-                        f'but the file {config_path} does not contain the '
-                        'account information.')
-        config['auth']['ssh_user'] = account.replace('@', '_').replace('.', '_')
+
+        subprocess.run(
+            'gcloud compute os-login ssh-keys add '
+            f'--key-file={public_key_path}',
+            check=True,
+            shell=True,
+            stdout=subprocess.DEVNULL)
+
+        # Try to get the os-login user from `gcloud`, as this is the most accurate way to figure out
+        # how this gcp user is meant to log in.
+        result = subprocess.run('gcloud compute os-login describe-profile', shell=True, stdout=subprocess.PIPE, universal_newlines=True)
```

Comment:
nit: we may want to explicitly add `--format json` (may need to use `json.loads` instead later or `--format yaml`), to make sure the users global config of the format does not affect this.